kernel/init.c

```diff
@@ -108,6 +108,15 @@ void __weak z_early_memcpy(void *dst, const void *src, size_t n)
 __boot_func
 void z_bss_zero(void)
 {
+	if (IS_ENABLED(CONFIG_ARCH_POSIX)) {
```

Comment:
this doesn't look like it belongs in this patch?

Reply:
It fixes a clang build error.  Really this was a historical mistake: it turns out that gcc's standard linker script emits __bss_start/_bss_end symbols that Zephyr was picking up and using here (needlessly, given that .bss is guaranteed cleared at the time of the anonymous mmap, but presumably harmlessly).   But it turns out that clang doesn't emit the latter, at least sometimes.  I don't know why, but it's not a standard thing and really it was just dumb luck that it ever worked at all.

Reply:
Sounds like you might just break this out into a separate PR? Dunno what the general sense around here is about 'one PR, one change'